src/ucs/memory/memtype_cache.c

```diff
@@ -71,20 +71,26 @@ static UCS_F_ALWAYS_INLINE ucs_memtype_cache_t *ucs_memtype_cache_get_global()
 }
 
 static UCS_F_ALWAYS_INLINE void
-ucs_memory_info_set_unknown(ucs_memory_info_t *mem_info)
+ucs_memory_info_set(ucs_memory_info_t *mem_info, ucs_memory_type_t mem_type,
+                    const void *address, size_t length)
 {
-    mem_info->type         = UCS_MEMORY_TYPE_UNKNOWN;
+    mem_info->type         = mem_type;
     mem_info->sys_dev      = UCS_SYS_DEVICE_ID_UNKNOWN;
-    mem_info->base_address = NULL;
-    mem_info->alloc_length = -1;
+    mem_info->base_address = (void*)address;
+    mem_info->alloc_length = length;
 }
 
-void ucs_memory_info_set_host(ucs_memory_info_t *mem_info)
+static UCS_F_ALWAYS_INLINE void
+ucs_memory_info_set_unknown(ucs_memory_info_t *mem_info, const void *address,
+                            size_t length)
 {
-    mem_info->type         = UCS_MEMORY_TYPE_HOST;
-    mem_info->sys_dev      = UCS_SYS_DEVICE_ID_UNKNOWN;
-    mem_info->base_address = NULL;
-    mem_info->alloc_length = -1;
+    ucs_memory_info_set(mem_info, UCS_MEMORY_TYPE_UNKNOWN, address, length);
+}
+
+void ucs_memory_info_set_host(ucs_memory_info_t *mem_info, const void *address,
+                              size_t length)
+{
+    ucs_memory_info_set(mem_info, UCS_MEMORY_TYPE_HOST, address, length);
```

Comment:
maybe no need for this wrappers? `ucs_memory_info_set` looks sufficient enough?

Reply:
fixed